File: src/types/attachment.ts
/** ファイル添付の種別 */
export type AttachmentType = 'text' | 'pdf' | 'image';

/** 添付ファイル情報 */
export interface Attachment {
  id: string;
  file_name: string;
  file_path: string;
  attachment_type: AttachmentType;
  size_bytes: number;
  /** テキスト/PDF抽出結果 */
  extracted_text?: string;
  /** 画像のBase64エンコード */
  base64_data?: string;
}

File: src-tauri/src/models/attachment.rs
use serde::{Deserialize, Serialize};

/// ファイル添付の種別
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum AttachmentType {
    Text,
    Pdf,
    Image,
}

/// 添付ファイル情報
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Attachment {
    pub id: String,
    pub file_name: String,
    pub file_path: String,
    pub attachment_type: AttachmentType,
    pub size_bytes: u64,
    /// テキスト/PDF抽出結果
    pub extracted_text: Option<String>,
    /// 画像のBase64エンコード
    pub base64_data: Option<String>,
}

/// 最大ファイルサイズ: 10MB
pub const MAX_FILE_SIZE: u64 = 10 * 1024 * 1024;
